### PR TITLE
ceph: suppress gosec errors for g204,g304,g101

### LINKED ISSUE #3765: Handle errors reported by gosec tool
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
gosec reports ~100 issues of various severity levels.

**Expected behavior:**
When running the gosec tool we should see a clean report, or else suppress the failures that we don't expect to be fixed. 

**How to reproduce it (minimal and precise):**

```
$GOPATH/bin/gosec pkg/... -severity=medium
...

Summary:
   Files: 449
   Lines: 67632
   Nosec: 0
  Issues: 96
```
Results are in attached log:
[gosec.log](https://github.com/rook/rook/files/3571682/gosec.log)

Some of these failures we do not need to worry about and can suppress. For example, there are several complaints about hard-coded creds that are just variable names.
```
[/Users/travisn/src/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/mgr/dashboard.go:41] - G101: Potential hardcoded credentials (Confidence: LOW, Severity: HIGH)
  > dashboardPasswordName          = "rook-ceph-dashboard-password"
```

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit suppress the gosec errors for

g204: Audit use of command execution
g304: File path provided as taint input
g101: Look for hard coded credentials

after this commit:-
`$GOPATH/bin/gosec -include=G204,G304,G101 cmd/... -severity=medium`

```
Summary:
   Files: 35
   Lines: 2955
   Nosec: 3
  Issues: 0
```
`$GOPATH/bin/gosec -include=G204,G304,G101 pkg/... -severity=medium`

```
Summary:
   Files: 519
   Lines: 82514
   Nosec: 17
  Issues: 0
```

Signed-off-by: subhamkrai <subhamkumarrai03@gmail.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #3765

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: cmd/rookflex/cmd/expander.go [hunk 1]
```diff
@@ -62,9 +63,9 @@ func handleExpandFs(cmd *cobra.Command, args []string) error {
 	var command *exec.Cmd
 	switch opts.FsType {
 	case "ext3", "ext4":
-		command = exec.Command("resize2fs", args[2])
+		command = exec.Command("resize2fs", args[2]) // #nosec G204
```
Comment (travisn) [inline cmd/rookflex/cmd/expander.go:66]: Whenever we suppress the gosec error, let's also add a description, and also put it on the line before. For example:
```
// #nosec G204 Rook controls the input to the exec arguments
command = exec.Command("xfs_growfs", "-d", args[2])
```

File: pkg/operator/ceph/webhook.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ limitations under the License.
 package operator
 
 import (
+	"os"
```
Comment (travisn) [inline pkg/operator/ceph/webhook.go:20]: you should be able to revert this `os` import change that moved it

File: cmd/rookflex/cmd/expander.go [hunk 3]
```diff
@@ -19,10 +19,11 @@ package cmd
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/rook/rook/pkg/daemon/ceph/agent/flexvolume"
-	"github.com/spf13/cobra"
 	"os/exec"
 	"strconv"
+
+	"github.com/rook/rook/pkg/daemon/ceph/agent/flexvolume"
```
Comment (travisn) [inline cmd/rookflex/cmd/expander.go:25]: These import changes can be reverted

File: cmd/rookflex/cmd/expander.go (hunk 1 above)
Comment (subhamkrai) [reply cmd/rookflex/cmd/expander.go]: Done.

File: pkg/operator/ceph/webhook.go (hunk 2 above)
Comment (subhamkrai) [reply pkg/operator/ceph/webhook.go]: Done.

File: cmd/rookflex/cmd/expander.go (hunk 3 above)
Comment (subhamkrai) [reply cmd/rookflex/cmd/expander.go]: Done.

File: pkg/operator/ceph/webhook.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ import (
 
 const (
 	appName                  = "rook-ceph-admission-controller"
-	secretVolumeName         = "webhook-certificates"
+	secretVolumeName         = "webhook-certificates" // #nosec G101 Rook controls the  hard coded credentials
```
Comment (travisn) [inline pkg/operator/ceph/webhook.go:34]:
```suggestion
	secretVolumeName         = "webhook-certificates" // #nosec G101 This is just a var name, not a real secret
```

Comment (subhamkrai) [reply pkg/operator/ceph/webhook.go]: Done.

File: pkg/daemon/ceph/agent/flexvolume/server.go [hunk 5]
```diff
@@ -258,6 +258,7 @@ func copyFile(src, dest string) error {
 	}
 	defer srcFile.Close()
 
+	// #nosec G304 Rook controls the File path provided as taint input
```
Comment (travisn) [inline pkg/daemon/ceph/agent/flexvolume/server.go:261]: I think it makes more sense on these comments without "taint"
```suggestion
	// #nosec G304 Rook controls the File path provided as input
```

File: pkg/operator/ceph/webhook.go (hunk 2 above)
Comment (travisn) [reply pkg/operator/ceph/webhook.go:20]: did the linter add this again? If you delete the blank line 21 it should fix this.

File: pkg/daemon/ceph/agent/flexvolume/server.go (hunk 5 above)
Comment (subhamkrai) [reply pkg/daemon/ceph/agent/flexvolume/server.go]: Done.

**Comment by subhamkrai** [comment]
actually linter just moved  `os` in top

**Pull request merged**